cmd/lncli/commands.go

```diff
@@ -707,7 +717,138 @@ func closeChannel(ctx *cli.Context) error {
 		req.ChannelPoint.OutputIndex = 0
 	}
 
-	stream, err := client.CloseChannel(ctxb, req)
+	txidChan := make(chan string, 1)
+	wg := sync.WaitGroup{}
+
+	wg.Add(1)
+	go func() {
+		printJSON(struct {
+			ClosingTxid string `json:"closing_txid"`
+		}{
+			ClosingTxid: <-txidChan,
+		})
+
+		wg.Done()
+	}()
+
+	if err := executeChannelClose(ctx, req, txidChan); err != nil {
+		return err
+	}
+
+	// We'll wait before exiting to make sure the closing txid is printed.
+	wg.Wait()
```

Comment:
I think we can accomplish a similar result without using the wait group. Waiting on `txidChan` should block to prevent the goroutine from returning. Why not just wait for the result, print json, then return?

Reply:
Yeah, I gave that a try. Basically, since `executeChannelClose` doesn't block without explicitly setting the `block` flag, we'll exit from `closeChannel` before being able to print the closing txid. I was able to get around that by using the `WaitGroup`, but maybe there's a better way?

Reply:
Ahh I see, without the wait group/go routine we wouldn't print the info until after a `block` call has been completed. Ignore my previous comment, I think this is a good workaround!